crates/agent/src/discovers.rs

```diff
@@ -124,31 +131,42 @@ impl DiscoverHandler {
             .await
             .context("clearing old errors")?;
 
+        let mut cmd = async_process::Command::new(format!("{}/flowctl-go", &self.bindir));
+        cmd.arg("api")
+            .arg("discover")
+            .arg("--config=/dev/stdin")
+            .arg("--image")
+            .arg(&image_composed)
+            .arg("--network")
+            .arg(&self.connector_network)
+            .arg("--output=json")
+            .arg("--log.level=warn")
+            .arg("--log.format=color");
+        if self.allow_local {
+            cmd.arg("--allow-local");
+        }
         let (discover, discover_output) = jobs::run_with_input_output(
             "discover",
             &self.logs_tx,
             row.logs_token,
             row.endpoint_config.0.get().as_bytes(),
-            async_process::Command::new(format!("{}/flowctl-go", &self.bindir))
-                .arg("api")
-                .arg("discover")
-                .arg("--config=/dev/stdin")
-                .arg("--image")
-                .arg(&image_composed)
-                .arg("--network")
-                .arg(&self.connector_network)
-                .arg("--output=json")
-                .arg("--log.level=warn")
-                .arg("--log.format=color"),
+            &mut cmd,
         )
         .await?;
 
         if !discover.success() {
+            let detail = if discover_output.is_empty() {
+                format!(
```

Comment:
I noticed that there was nothing in `log_lines` when this happened, and that was pretty disheartening. So this was just an unrelated improvement for that case.